src/components/modal/ModalCreateBooks.tsx

```diff
@@ -0,0 +1,188 @@
+  })
+
+  const [ authorData, setAuthorData ] = useState<Author>({
+    Id: 0,
+    FirstName: '',
+    LastName: ''
+  })
+
+
+  const handleImgUpload = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
+        reader.readAsDataURL(files[0])
+        setFileImg(files[0])
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data) setCoverImg(base64data as string)
+        }
+      }
+    }
+  }
+  const hanlderCreateBooks = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('Title', booksData.Title)
+      form.append('Description', booksData.Description)
+      form.append('Isbn', booksData.Isbn)
+      form.append('Quantity', booksData.Quantity.toString())
+      form.append('Cover', fileImg)
+      form.append('PublishDate', booksData.PublishDate)
+      booksData.AuthorIds.forEach((author) => form.append('AuthorIds', author.Id.toString()))
+      await BookService.createBook(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handleCreateAuthor = async (event: FormEvent<HTMLFormElement>) => {
+    handlerNewAuthor()
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('FirstName', authorData.FirstName)
+      form.append('LastName', authorData.LastName)
+      await AuthorService.createAuthor(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handlerNewAuthor = () => {
+    AuthorService.getAuthors().then((response) => {
+      setAtuhors(response)
+    })
+    setShowAddAuthor(true)
+    setShowAddButton(true)
+  }
+
+  const handlerCloseModal = () => {
+    onClose()
+  }
+
+  const handleFormAuthor = () => {
+    setShowAddAuthor(false)
+    setShowAddButton(false)
+  }
+
+  const handleCloseFormAuthor = () => {
+    setShowAddAuthor(true)
+    setShowAddButton(true)
+  }
+
+  useEffect(() => {
+    AuthorService.getAuthors()
+      .then(response => {
+        setAtuhors(response)
+      })
+  }, [])
+
+  const handleAuthor = (newAuthors: MultiValue<Author>) => {
+    setBooksData((prev) => ({ ...prev, AuthorIds: newAuthors as Author[] }))
+  }
+
+  if(!open) return null
+
+
+  return ReactDOM.createPortal(
+    <div className='overlay'>
+      <div className='content'>
+        <form className='form_for_modal' onSubmit={hanlderCreateBooks}>
+          <div className='heder_modal'>
+            <button type='button' onClick={handlerCloseModal} className='x_button_for_modal'><XIconForModal className='x_icon_for_modal'/></button>
```

Comment:
if handler is constructed of one line, that just updates the state we can write it as arrow function inside onClick

Reply:
I fix this.